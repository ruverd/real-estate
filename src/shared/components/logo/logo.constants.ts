export const LOGO_TEXT = 'Real Estate';