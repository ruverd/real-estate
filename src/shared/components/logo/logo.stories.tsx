import type { Meta, StoryObj } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";

import { Logo } from "./logo";

const meta: Meta<typeof Logo> = {
  title: "Components/Logo",
  component: Logo,
  tags: ["autodocs"],
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Logo>;

export const Default: Story = {
  render: () => <Logo />,
};

export const InContainer: Story = {
  render: () => (
    <div className="bg-background p-4 rounded-md shadow-sm">
      <Logo />
    </div>
  ),
};

export const InNavbar: Story = {
  render: () => (
    <div className="bg-background p-4 border-b flex items-center">
      <Logo />
      <div className="ml-auto flex gap-4">
        <span className="text-sm">Home</span>
        <span className="text-sm">Properties</span>
        <span className="text-sm">Contact</span>
      </div>
    </div>
  ),
}; 