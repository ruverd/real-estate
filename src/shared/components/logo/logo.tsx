import { House } from "lucide-react";
import { Link } from "react-router-dom";

import { LOGO_TEXT } from "./logo.constants";

export const Logo = () => {
  return (
    <Link to="/">
      <div className="flex items-center gap-1">
        <House className="size-6" />
        <span className="font-sans font-semibold leading-none text-lg text-foreground">{LOGO_TEXT}</span>
      </div>
    </Link>
  );
};
