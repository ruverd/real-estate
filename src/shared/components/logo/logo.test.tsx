import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Logo } from "./logo";
import { LOGO_TEXT } from "./logo.constants";

const renderLogo = () => {
  return render(
    <BrowserRouter>
      <Logo />
    </BrowserRouter>
  );
};

describe("Logo", () => {
  it("should render the logo component", () => {
    renderLogo();
  
    expect(screen.getByText(LOGO_TEXT)).toBeInTheDocument();
  });

  it("should contain a link to the home page", () => {
    renderLogo();
    
    const link = screen.getByRole("link");

    expect(link).toHaveAttribute("href", "/");
  });

  it("should contain the House icon", () => {
    renderLogo();
    
    const iconContainer = screen.getByText(LOGO_TEXT).parentElement;

    expect(iconContainer).toBeInTheDocument();
    expect(iconContainer?.querySelector("svg")).toBeInTheDocument();
  });

  it("should have the correct styling classes", () => {
    renderLogo();
    
    const container = screen.getByText(LOGO_TEXT).parentElement;

    expect(container).toHaveClass("flex items-center gap-1");
    
    const logoText = screen.getByText(LOGO_TEXT);

    expect(logoText).toHaveClass("font-sans font-semibold leading-none text-lg text-foreground");
  });

  it("should match snapshot", () => {
    const { container } = renderLogo();

    expect(container.firstChild).toMatchSnapshot();
  });
});
