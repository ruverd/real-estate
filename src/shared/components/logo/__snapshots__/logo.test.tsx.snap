// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > should match snapshot 1`] = `
<a
  data-discover="true"
  href="/"
>
  <div
    class="flex items-center gap-1"
  >
    <svg
      class="lucide lucide-house size-6"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
      />
      <path
        d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
      />
    </svg>
    <span
      class="font-sans font-semibold leading-none text-lg text-foreground"
    >
      Real Estate
    </span>
  </div>
</a>
`;
